torchtune/datasets/_chat.py

```diff
@@ -10,7 +10,7 @@
 
 from datasets import load_dataset
 from torch.utils.data import Dataset
-from torchtune.config._utils import _get_chat_format
```

Comment:
I guess this gets to the heart of half of the comments I left earlier. My concern with this is we're creating two ways to do the same things in the config. Is this meant to be a way to do nested instantiation from yaml without supporting recursive instantiation?

Reply:
> Is this meant to be a way to do nested instantiation from yaml without supporting recursive instantiation?

ding ding ding

Reply:
it's a fair concern. this is my workaround for recursive instantiation for now... may need to revisit this if lack of recursive instantiation becomes increasingly problematic, especially as datasets become more complex with multimodal and iterable. One of the reason we did not want to use hydra was that recursive instantiation is a bit too powerful and may lead to coding via config (cc @ebsmothers). Happy to chat more offline if you want to go into that rabbit hole :)